solr/src/main/java/com/yahoo/ycsb/db/SolrDBClient.java

```diff
@@ -0,0 +1,339 @@
+/**
+ * Copyright (c) 2016 YCSB contributors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.db;
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.UpdateResponse;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrInputDocument;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+
+/**
+ * Solr client for YCSB framework.
+ *
+ * <p>
+ * Default properties to set:
+ * </p>
+ * <ul>
+ * <li>table = ycsb
+ * <li>solr.cloud = false
+ * <li>solr.host.list
+ * <li>solr.commit.within.time
+ * <li>solr.batch.mode
+ * <li>zookeeper.host.list
+ * </ul>
+ *
+ * @author Gerard Haughian
+ *
+ */
+public class SolrDBClient extends DB {
+
+  public static final String DEFAULT_COLLECTION_NAME = "ycsb";
+  public static final String DEFAULT_CLOUD_MODE = "false";
+  public static final String DEFAULT_BATCH_MODE = "false";
+  public static final String DEFAULT_REMOTE_ZOOKEEPER_HOST = "localhost:2181";
+  public static final String DEFAULT_REMOTE_SOLR_HOST = "localhost:8983";
+  public static final String DEFAULT_COMMIT_WITHIN_TIME = "1000";
+
+  private SolrClient client;
```

Comment:
if this is the only place where the solr object is used, you could instead refer to it by the full qualified package name here and then you wouldn't have to worry about your class name colliding.